src/types/platforms-types/multi-track-multiprint-contract.ts

```diff
@@ -83,13 +77,13 @@ const mapArtistProfile = ({ apiTrack, nft, contract }: { apiTrack: any, nft?: NF
   }
 };
 
-const mapNFTtoTrackID = (nft: NFT): string => {
-  const trackName = getTrait(nft, 'Track');
+const mapNFTtoTrackID = (nft: NFT, extractor: any): string => {
+  const trackName = strictSlugify(extractor(nft));
   return `ethereum/${formatAddress(nft.contractAddress)}/${trackName}`;
 };
 
-const mapNFTsToTrackIds = async (nfts: NFT[]): Promise<{ [trackId: string]: NFT[] }> => {
-  return _.groupBy(nfts, nft => mapNFTtoTrackID(nft));
+const mapNFTsToTrackIds = async (nfts: NFT[], dbClient?: DBClient | undefined, extractor?: any): Promise<{ [trackId: string]: NFT[] }> => {
+  return _.groupBy(nfts, nft => mapNFTtoTrackID(nft, extractor));
```

Comment:
lets make extractor a specific type and throw error if its undefined

Reply:
its also seems kinda weird that the mapNFTsToTrackIds function cares about a titleExtractor - seems like bad seperation of concerns, especially since this logic is very specific to this type rather than more general. i think the function should just take in the nftFactory and handle the rest internally?

Reply:
I was also pretty uncomfortable with this, it feels a bit messy. 

But the reason for passing the `extractor` instead of the `nftFactory` is that this function works on multiple NFTs, which (as I understand it) might not all map to the same nftFactory. 

If that's the correct assumption, then passing the NFT along with the function that operates on it to get its title is slightly better than locking down all the NFTs to a single nftFactory.

Reply:
I take it there isn't an easy way to get the `nftFactory` that created the `NFT` without querying the DB, right @musnit ? That would clean up the function signature.

Reply:
> But the reason for passing the `extractor` instead of the `nftFactory` is that this function works on multiple NFTs, which (as I understand it) might not all map to the same nftFactory.

ah yeah that's a good point. i suppose by that same logic they may not even map to the same extractor, so ideally we should provide the extractor per-nft

> If that's the correct assumption, then passing the NFT along with the function that operates on it to get its title is slightly better than locking down all the NFTs to a single nftFactory.
yeah exactly - as you say